src/macaron/slsa_analyzer/git_url.py

```diff
@@ -566,6 +566,33 @@ def get_remote_vcs_url(url: str, clean_up: bool = True) -> str:
     return url_as_str
 
 
+def clean_url(url: str) -> urllib.parse.ParseResult | None:
```

Comment:
Because we have refactored this logic into its own function. I think it's good if we add some unit tests for it (if it's not out of scope for this PR).

Reply:
Hmm, to me at least, this may not be necessary though. This function is extracted from the `parse_remote_url` function, which by itself does not have any unit tests and it is so far OK because this is a utility function called by others, each of which has unit tests already. So if we didn't have unit tests for `parse_remote_url` before and were OK with it, I wouldn't think we need unit tests for this one.

Reply:
I think @benmss has already added some unit tests :sweat_smile: . 
I agree with @nathanwn that unit tests are not totally necessary as this logic has been tested through unit tests of other functions that call it. However, I don't think we should revert what @benmss has added as it doesn't cause extra issues (for me). Please let me know what you think @nathanwn.   
With that being said, I will approve the changes.

Reply:
Indeed, I added some before seeing the follow up message. Probably there is no harm in keeping them now that they exist :)

Reply:
Let's keep the test.